talk/owt/sdk/p2p/p2ppeerconnectionchannel.cc

```diff
@@ -1050,11 +1037,14 @@ void P2PPeerConnectionChannel::DrainPendingStreams() {
   RTC_LOG(LS_INFO) << "Draining pending stream";
   ChangeSessionState(kSessionStateConnecting);
   bool negotiation_needed = false;
+  rtc::scoped_refptr<webrtc::PeerConnectionInterface> temp_pc_ = GetPeerConnectionRef();
   // First to publish the pending_publish_streams_ list.
   {
     std::lock_guard<std::mutex> lock(pending_publish_streams_mutex_);
     for (auto it = pending_publish_streams_.begin();
          it != pending_publish_streams_.end(); ++it) {
+      if (!temp_pc_)
```

Comment:
why not just return early at the top of the function before getting lock?

Reply:
We need to make sure the pending_ vectors are cleared, since they could leak SHM clients even after the connection is closed. Rather than have a separate return, I just let DrainPendingStreams() skip the peer connection operations but still clear the vectors one last time.

Reply:
This must be done inside the lock so there is no Publish() or Unpublish() call running concurrently that might be adding to the vectors.